Duplicate prefix error in data update

The latest data updater job is failing due to the duplicate URI prefix `https://www.chemspider.com/search?params=` between `inchi` and `inchikey` which is presumably due to an update in one of the external sources.

```
            if duplicate_uri_prefixes:
>               raise DuplicateURIPrefixes(duplicate_uri_prefixes)
E               curies.api.DuplicateURIPrefixes: Duplicate URI prefixes:
E               
E               https://www.chemspider.com/search?params=:
E               	prefix='inchi' uri_prefix='https://www.ebi.ac.uk/chebi/advancedSearchFT.do?searchString=' prefix_synonyms=[] uri_prefix_synonyms=['http://bioregistry.io/inchi:', 'http://chemapps.stolaf.edu/jmol/jmol.php?&model=', 'http://identifiers.org/inchi/', 'http://identifiers.org/inchi:', 'http://n2t.net/inchi:', 'http://rdf.openmolecules.net/?', 'http://webbook.nist.gov/cgi/cbook.cgi?', 'http://www.chemspider.com/search?params=', 'http://www.ebi.ac.uk/chebi/advancedSearchFT.do?searchString=', 'https://bioregistry.io/inchi:', 'https://chemapps.stolaf.edu/jmol/jmol.php?&model=', 'https://identifiers.org/inchi/', 'https://identifiers.org/inchi:', 'https://n2t.net/inchi:', 'https://rdf.openmolecules.net/?', 'https://webbook.nist.gov/cgi/cbook.cgi?', 'https://www.chemspider.com/search?params='] pattern='^InChI\\=1S?\\/[A-Za-z0-9\\.]+(\\+[0-9]+)?(\\/[cnpqbtmsih][A-Za-z0-9\\-\\+\\(\\)\\,\\/\\?\\;\\.]+)*$'
E               	prefix='inchikey' uri_prefix='https://www.chemspider.com/search?params=' prefix_synonyms=[] uri_prefix_synonyms=[] pattern='^[A-Z]{14}\\-[A-Z]{10}(\\-[A-Z])?$'

.tox/py/lib/python3.12/site-packages/curies/api.py:866: DuplicateURIPrefixes
```

See https://github.com/biopragmatics/bioregistry/actions/runs/16383831033/job/46300074653

--- a/src/bioregistry/external/miriam/__init__.py
+++ b/src/bioregistry/external/miriam/__init__.py
@@ -62,12 +62,13 @@ def get_miriam(
     return rv
 
 
-#: Pairs of MIRIAM prefix and provider codes to skip
+#: Pairs of MIRIAM prefix and provider codes (or name, since some providers don't have codes)
 PROVIDER_BLACKLIST = {
     ("ega.study", "omicsdi"),
     # see discussion at https://github.com/biopragmatics/bioregistry/pull/944
     ("bioproject", "ebi"),
     ("pmc", "ncbi"),
+    ("inchi", "InChI through Chemspider"),
 }
 
 
@@ -104,7 +105,11 @@ def _process(record: dict[str, Any]) -> dict[str, Any]:
     extras = []
     for provider in rest:
         code = provider["code"]
-        if code in SKIP_PROVIDERS or (prefix, code) in PROVIDER_BLACKLIST:
+        if (
+            code in SKIP_PROVIDERS
+            or (prefix, code) in PROVIDER_BLACKLIST
+            or (prefix, provider["name"]) in PROVIDER_BLACKLIST
+        ):
             continue
         del provider["official"]
         extras.append(provider)

